common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -3034,11 +3065,18 @@ def _get_parents_from_structure(self, block_key, structure):
         Given a structure, find block_key's parent in that structure. Note returns
         the encoded format for parent
         """
-        return [
-            parent_block_key
-            for parent_block_key, value in structure['blocks'].iteritems()
-            if block_key in value.fields.get('children', [])
-        ]
+        cache_key = u'structure.{structure_id}'.format(structure_id=structure['_id'])
+        children_to_parents = cache.get(cache_key)
```

Comment:
And also because you'd be introducing a Django dependency. Maybe just try it first without the caching and see where we end up in terms of profiling times? It's possible that the network hop and deserialization wouldn't be worth it when compared to a single traversal of a structure we already have in memory.